utils/b2i/wmo_codes.py

```diff
@@ -0,0 +1,207 @@
+'''
+Understanding WMO Identifiers for Tropical Moorings
+
+    WMO Identifiers:
+        Old Format (5-digit): Used historically for ocean platforms (e.g., moored buoys, drifting buoys) in the form A1bwnnn, where:
+            A1: Region (e.g., 4 for Asia, 7 for North America, 8 for South America).
+            bw: Platform type or subregion (e.g., 6 for moored buoys in some contexts).
+            nnn: Station number (000–999).
+            Example: 64787 (A1=6, bw=4, nnn=787) for a moored buoy.
+        New Format (7-digit): Introduced to expand identifier capacity,
+        especially for BUFR (Binary Universal Form for the Representation
+        of meteorological data). The format is A1bwnnnnn, where:
+            A1: Region (same as above).
+            bw: Platform type/subregion.
+            nnnnn: Station number (00000–99999, up to 5 digits).
+            Conversion rule: Most 5-digit IDs are mapped to 7-digit by inserting “00” in the 3rd and 4th positions (e.g., 64787 → 6400787).
+        WIGOS Station Identifiers (WSI): Modern WMO identifiers
+        use a 4-block structure (e.g., 0-22000-0-3100007 for
+        a USA moored buoy in WMO area 31).
+        The 7-digit format aligns with BUFR reporting but
+        may not always map directly to WSI’s local identifier block.
+    Tropical Moorings:
+        These are moored buoys in arrays like TAO/TRITON (Pacific),
+        PIRATA (Atlantic), or RAMA (Indian Ocean), reporting via
+        GTS in BUFR format (e.g., IOBX08 KPML).
+        WMO numbers for modern T-Flex moorings (replacing ATLAS)
+        use 7-digit analogs of older 5-digit codes at the same
+        site (e.g., 23010 → 2300010 for a RAMA mooring at 4°S 81°E)
+'''
+
+
+# Function to convert 5-digit to 7-digit WMO code
+def convert_5_to_7_digit(wmo_5digit):
+    """
+    If the code is already 7 digits, it is returned as is.
+    Convert a 5-digit WMO code (A1bwnnn) to 7-digit (A1bwnnnnn)
+    by inserting '00' after bw.
+    Returns the 7-digit code as a string.
+    """
+    if len(wmo_5digit) == 7:
+        return wmo_5digit
+    if not wmo_5digit.isdigit() or len(wmo_5digit) != 5:
+        return None  # Invalid 5-digit code
+    # Extract A1, bw, nnn
+    a1 = wmo_5digit[0]
+    bw = wmo_5digit[1]
+    nnn = wmo_5digit[2:]
+    # Convert to 7-digit: A1 + bw + '00' + nnn
+    wmo_7digit = f"{a1}{bw}00{nnn}"
+    return wmo_7digit
+
+
+TAO_TRITON = [
+    4300011,

```

Comment:
As before, if these ever change we should aim towards moving them to config files